### ISSUE TITLE
Enhancement: Add a console.warn to experimental-utils

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#5982]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Relevant Package

utils

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

The `@typescript-eslint/utils` package used to be called `@typescript-eslint/experimental-utils`. We procrastinated removing the `experimental-` from the name for a while, then finally did it in https://github.com/typescript-eslint/typescript-eslint/issues/4139. We left the old `experimental-` package behind so as to not break things for people. But, it'd be good for them to stop referring to the old package name.

See this line in `experimental-utils` README.md:

https://github.com/typescript-eslint/typescript-eslint/blob/91d71bcf6b2f822116d25b281e289fe1d5005401/packages/experimental-utils/README.md#L21

Now that we're working on breaking changes for our next major version, v6, we can add that `console.warn` in!

Proposal: in https://github.com/typescript-eslint/typescript-eslint/blob/91d71bcf6b2f822116d25b281e289fe1d5005401/packages/experimental-utils/src/index.ts, add a `console.warn` that says something like the text from the README.md:

https://github.com/typescript-eslint/typescript-eslint/blob/91d71bcf6b2f822116d25b281e289fe1d5005401/packages/experimental-utils/README.md#L13-L14

### Fail

```typescript
n/a
```

### Pass

```typescript
n/a
```

### Additional Info

This is a breaking change and will need to target the `v6` branch.

**Comment by JoshuaKGoldberg** [comment]
Fixed in #6179.
